scipy/stats/tests/test_stats.py

```diff
@@ -7917,6 +7917,222 @@ def test_alias(self):
+        UB = [int(N[i])-1-tmp[i] if (not np.isnan(tmp[i])) else np.nan
+            for i in range(m)]
+        LB_val = [ x[i][LB[i]] if (not np.isnan(LB[i])) else np.nan 
+                    for i in range(m)]
+        UB_val = [ x[i][UB[i]] if (not np.isnan(UB[i])) else np.nan 
+                for i in range(m)]
+        test_val = [(LB_val[i], UB_val[i]) for i in range(m)]
+
+        public = [(
+            stats.quantile_test(x[i], 
+                                0, 
+                                quantile_def[i], 
+                                alternative='greater'
+                                ).confidence_interval(confidence[i]).low,
+            stats.quantile_test(x[i], 
+                                0, 
+                                quantile_def[i], 
+                                alternative='less'
+                                ).confidence_interval(confidence[i]).high)
+            for i in range(m)]
+        assert_equal(public, test_val)
+
+        # Test for the two-sided case
+        # .. Two-sided CI are built of two one-sided CIs 
+        # with larger confidence
+        confidence_two_sided = (1 + confidence)/2
+        tmp = [confint_naive(int(N[i]), 
+                             quantile_def[i], 
+                             confidence_two_sided[i])
+            for i in range(m)]
+        LB = [i if (not np.isnan(i)) else np.nan for i in tmp]
+        # .. UB is (N-1-LB(1-q))
+        tmp = [confint_naive(int(N[i]), 
+                             1-quantile_def[i], 
+                             confidence_two_sided[i])
+            for i in range(m)]
+        UB = [int(N[i])-1-tmp[i] if (not np.isnan(tmp[i])) else np.nan
+            for i in range(m)]
+        LB_val = [ x[i][LB[i]] if (not np.isnan(LB[i])) else np.nan 
+                for i in range(m)]
+        UB_val = [ x[i][UB[i]] if (not np.isnan(UB[i])) else np.nan 
+                for i in range(m)]
+        test_val = [(LB_val[i], UB_val[i]) for i in range(m)]
+        # .. Compare to the public function
+        tmp = [
+            stats.quantile_test(x[i], 
+                                0, 
+                                quantile_def[i], 
+                                alternative='two-sided'
+                                ).confidence_interval(confidence[i])
+            for i in range(m)]
+        public = [(tmp[i].low, tmp[i].high) for i in range(m)]
+        assert_equal(public, test_val)
+
+    def test_R_ci_quantile(self):
+        # Using the same test cases as the ci_quantile R function
+        # https://github.com/mayer79/confintr/blob/main/tests/testthat/test-ci_measures_of_location.R#L95
+
+        x = np.exp(np.arange(0, 1.01, 0.01))
+
+        lb, ub = stats.quantile_test(x, 
+                                     0, 
+                                     0.4, 
+                                     alternative='two-sided'
+                                     ).confidence_interval(0.95)
```

Comment:
For readability, here and elsewhere, please refactor:
```suggestion
        res = stats.quantile_test(x, 0, 0.4, alternative='two-sided')
        lb, ub = res.confidence_interval(0.95)
```

Reply:
ack, will do